libs/algorithms/include/algorithms/reachability/verdict.hpp

```diff
@@ -0,0 +1,65 @@
+
+#ifndef MINIMC_VERDICT_HPP
+#define MINIMC_VERDICT_HPP
+
+#include "algorithms/algorithms.hpp"
+#include "cpa/interface.hpp"
+#include <variant>
+
+
+namespace MiniMC {
+namespace Algorithms {
+namespace Reachability {
+namespace Verdict {
+
+enum class Verdict { Found, NotFound };
+
+template<Verdict V>
+struct VerdictData{
+  static const bool isFound = false;
+};
+
+struct FoundContent {
+  MiniMC::CPA::AnalysisState& state;
```

Comment:
I'm not a fan of having the Result keeping a reference to the reachability found-attribute. If it does, then the Result can only be used  as long as reachability algorithm is "live" --- and then we might as well just access the the found attribute directly.

Reply:
It made no sense to me to return only Found/NotFound, but i overlooked that the found state is saved in the Reachability instance and now I wonder if the Result class is obsolete. I agree with this should not be a reference.